File: smallest-difference/SmallestDifference.cs
using System;

public class SmallestDifference {
    public static int[] smallestDifference(int[] arrayOne, int[] arrayTwo) {
        Array.Sort(arrayOne);
        Array.Sort(arrayTwo);
        int idxOne = 0;
        int idxTwo = 0;
        int smallest = int.MaxValue;
        int current = int.MaxValue;
        int[] result = new int[2];
        while(idxOne < arrayOne.Length && idxTwo < arrayTwo.Length) {
            int first = arrayOne[idxOne];
            int second = arrayTwo[idxTwo];
            if(first < second) {
                current = second - first;
                idxOne++;
            } else if (second < first) {
                current = first - second;
                idxTwo++;
            } else {
                return new int[] { first, second };
            }
            if(current < smallest) {
                smallest = current;
                result = new int[]{ first, second };
            }
        }
        return result;
    }
}

File: smallest-difference/SmallestDifference.swift
import Foundation

class SmallestDifference {
    func smallestDifference(arrayOne: [Int], arrayTwo: [Int]) -> [Int] {
        arrayOne.sort(); arrayTwo.sort()
        var result = []
        var idxOne = 0
        var idxTwo = 0
        var smallest = Int.max
        var current = Int.max
        while(idxOne < arrayOne.count && idxTwo < arrayTwo.count) {
            var first = arrayOne[idxOne]
            var second = arrayTwo[idxTwo]
            if(first < second) {
                current = second - first
                idxOne += 1
            } else if (first > second) {
                current = first - second
                idxTwo += 1
            } else {
                return [ first, second ]
            }
            if(current < smallest) {
                smallest = current
                result = [ first, second ]
            }
        }
        return result
    }
}